clang/lib/Basic/Targets.cpp

```diff
@@ -833,6 +833,22 @@ TargetInfo::CreateTargetInfo(DiagnosticsEngine &Diags,
   if (!Target->handleTargetFeatures(Opts->Features, Diags))
     return nullptr;
 
+  // If TuneCPU is set, check if it contains all instruction sets needed by
+  // current feature map.
+  if (!Opts->TuneCPU.empty() && Opts->TuneCPU != Opts->CPU) {
```

Comment:
Can this handle the case that we use default cpu(i.e. no -mcpu set)?

Reply:
If `TuneCPU` is empty, this check won't happen. If no `-mcpu` set, `CPU` here depends on triple: for `powerpc64le-unknown-linux-gnu`, that's `ppc64le` (not `pwr8`); for `powerpc64-ibm-aix`, that's `pwr7` (not `ppc64`). Here the comparison is reliable.